### PR TITLE
Refactor confirm conditions views to make way for individual conditions

### PR DESCRIPTION
**Pull request by benswannack**
## Context
We are going to put a new flow behind a feature flag alongside the current confirm conditions flow.

## Changes proposed in this pull request
Refactor confirm conditions views into components to make future code read better

## Trello card
https://trello.com/c/4keyF4am/3791-implement-individual-conditions-ui-in-manage-for-non-deferred-offers

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: app/components/provider_interface/conditions_component.html.erb [hunk 1]
```diff
@@ -11,13 +11,7 @@
...
             <td class="govuk-table__cell"><%= condition.text %></td>
             <td class="govuk-table__cell govuk-table__cell--numeric">
-              <% if condition.met? %>
-                <%= govuk_tag(text: 'Met', colour: 'green') %>
-              <% elsif condition.unmet? %>
-                <%= govuk_tag(text: 'Not met', colour: 'red') %>
-              <% else %>
-                <%= govuk_tag(text: 'Pending', colour: 'grey') %>
-              <% end %>
+              <%= render ProviderInterface::ConditionStatusTagComponent.new(condition) %>
```
Comment (benswannack) [inline app/components/provider_interface/conditions_component.html.erb:14]: These places should really be using the same conditions list component. It's a component that's needed in the next phase of the work too, so I'll add it in this PR too

Comment (benswannack) [reply app/components/provider_interface/conditions_component.html.erb:14]: We can address this after the individual conditions work is done. I'll add a component to handle this and we can refactor old code to use the new component

File: app/components/provider_interface/condition_status_bulk_review_page_component.html.erb [hunk 2]
```diff
@@ -0,0 +1,40 @@
+<%= form_with(
+  model: form_object,
+  url: provider_interface_application_choice_update_conditions_path(@application_choice.id),
```
Comment (despo) [inline app/components/provider_interface/condition_status_bulk_review_page_component.html.erb:3]:
```suggestion
  url: provider_interface_application_choice_update_conditions_path(@application_choice),
```

File: app/components/provider_interface/condition_status_bulk_edit_form_component.html.erb [hunk 3]
```diff
@@ -0,0 +1,35 @@
+<%= form_with(
+  model: form_object,
+  url: provider_interface_application_choice_confirm_update_conditions_path(application_choice.id),
```
Comment (despo) [inline app/components/provider_interface/condition_status_bulk_edit_form_component.html.erb:3]:
```suggestion
  url: provider_interface_application_choice_confirm_update_conditions_path(application_choice),
```

File: app/components/provider_interface/condition_status_bulk_review_page_component.html.erb [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  <%= f.hidden_field :conditions_met %>
+
+  <% if form_object.conditions_met? %>
+    <%= f.govuk_submit 'Confirm they have met your conditions' %>
+  <% else %>
+    <%= f.govuk_submit 'Confirm they have not met your conditions', warning: true %>
+  <% end %>
+
+  <p class="govuk-body">
+    <%= govuk_link_to 'Cancel', provider_interface_application_choice_path(application_choice.id), class: 'govuk-link--no-visited-state' %>
```
Comment (despo) [inline app/components/provider_interface/condition_status_bulk_review_page_component.html.erb:38]:
```suggestion
    <%= govuk_link_to 'Cancel', provider_interface_application_choice_path(application_choice), class: 'govuk-link--no-visited-state' %>
```

File: app/components/provider_interface/condition_status_tag_component.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+module ProviderInterface
+  class ConditionStatusTagComponent < ViewComponent::Base
+    include ViewHelper
```
Comment (despo) [inline app/components/provider_interface/condition_status_tag_component.rb:3]: I don't think this is needed

File: app/components/provider_interface/condition_status_bulk_edit_form_component.html.erb [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  method: :patch,
+) do |f| %>
+  <%= f.govuk_error_summary %>
+
+  <h1 class="govuk-heading-xl">
+    Confirm if the candidate has met all conditions
+  </h1>
+
+  <%= render SummaryListComponent.new(rows: [
```
Comment (despo) [inline app/components/provider_interface/condition_status_bulk_edit_form_component.html.erb:12]: Can we generate this hash in the component's ruby file.

**Review by despo — APPROVED** [review]

File: app/components/provider_interface/condition_status_bulk_edit_form_component.html.erb (hunk 6 above)
Comment (benswannack) [reply app/components/provider_interface/condition_status_bulk_edit_form_component.html.erb]: 👍  Also addressed in the other component, thanks!

**Review by cwrw — COMMENTED** [review]
A bit late but looks good, only suggestion would be to add some specs for the other components, but if we're editing later should be ok

**Pull request merged**